solarwinds_apm/apm_config.py

```diff
@@ -549,7 +550,35 @@ def update_transaction_filters(self, cnf_dict: dict) -> None:
                 cfilter["regex"]
                 for cfilter in self.__config["transaction_filters"]
             ]:
-                self.__config["transaction_filters"].append(filter)
+                txn_filter = {}
+                txn_filter[
+                    "tracing_mode"
+                ] = OboeTracingMode.get_oboe_trace_mode(filter["tracing"])
+
+                if not isinstance(filter["regex"], str):
+                    logger.warning(
+                        "Transaction filter regex must be string or regex. Ignoring: %s",
+                        filter,
+                    )
+                    continue
+
+                if not len(filter["regex"]) > 0:
+                    logger.warning(
+                        "Transaction filter regex must not be empty. Ignoring: %s",
+                        filter,
+                    )
+                    continue
+
+                try:
+                    re.compile(filter["regex"])
```

Comment:
Since you're already compiling the regex in here wouldn't it make sense to store the compiled version in `txn_filter` so it doesn't have to get recompiled on every match

Reply:
Good idea! Changed in [d28c040](https://github.com/solarwindscloud/solarwinds-apm-python/pull/136/commits/d28c040a158f4349bb918285dd92e04fda31cc80).